### DISCUSSION TITLE
Hono server on GitHub Pages

### DISCUSSION THREAD

**Discussion by xkeshav** [honojs/hono#2865] [General]
I have to present on a confrence and I want to showcase the hono API example using various endpoints

GitHub Pages are great to deploy in minutes but when i try it; it load just readmd.md of my repo( which seeme correct as it is lansing page )

but what i want is that ; likewise in local I can run devleopment in localhost:3000

can someone help me how to enable it with build script and it runs over github.io pages so that if I use

github.io/repo_name/<API_Endpoint> then it load that page

is that possible? what changes need to add in workflow or what?  

[this is my hono based repo](https://github.com/xkeshav/o_ho_no)

[see github pages](https://xkeshav.github.io/o_ho_no/)

please guide how can we achive ?

**Comment by NicoPlyley** [comment]
Hi @xkeshav,
I don't believe this is possible with Hono. As far as I know, GitHub Pages only supports static HTML, CSS, and JS or will do a build step to generate static content. You can use CloudFlare Pages which has a very generous free tier.

**Reply by xkeshav** [reply]
Oh, Thanks

I was trying and added a workflow now that workflow keeps running

**Reply by xkeshav** [reply]
Can you please share any reference link then it would be great as cloudflare is too large to explore.
Moreover can we utilize honoX ?

**Reply by NicoPlyley** [reply]
Here is an example on CF Workers
https://github.com/NicoPlyley/hono-auth
And how to use it
https://www.youtube.com/watch?v=H7Qe96fqg1M _It's a long video but you only need the first bit_

I'm happy to answer any questions if you get stuck. You can also try the [Discord](https://discord.gg/KMh2eNSdxV) for quick questions. 

I am not sure if HonoX would work in something like GitHub Pages

**Reply by xkeshav** [reply]
Thanks for the reply.
I have gone through the wrangler.toml

most of lines are commented and only database connection has been configured 

So my question here is

I do not have any database so far, I just want to showcase how hono API works so creates few endpoints and paths

Then how and what do I configure in .toml file

**Reply by NicoPlyley** [reply]
When you create a project you'll get a wrangler.toml file with everything commented out. That's for using internal services cloud flare provides like the database or kv storage. You don't need any of them so you can just ignore it

**Reply by NicoPlyley** [reply]
You will need the name and compatibility date though. Leave the date the same, change the name to your projects name
